modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -63,7 +64,7 @@ class LsfExecutor extends AbstractGridExecutor {
         super.init()
         queueInterval = session.getQueueStatInterval(name)
         log.debug "Creating executor '$name' > queue-stat-interval: ${queueInterval}"
-        def memoryUnitLSF = getLSFmemoryUnits() ? : 'MB' 
+        def memoryUnitLSF = LsfExecutor.getLSFmemoryUnits() ? : 'MB' 
```

Comment:
`LsfExecutor.getLSFmemoryUnits()` is not correct because that implies a static method definition and `getLSFmemoryUnits` should not br static. 

The operator is `?:` not `? :` as `+ =` is not the same same `+=`  .. and I guess that's why it's failing. 

Last, I was wrong to suggest you to implement this logic in the `init` method, it should be `register` instead (always a template method in the superclass).

Reply:
Thank you for the feedback. These comments make a good deal of sense. 

> Last, I was wrong to suggest you to implement this logic in the `init` method, it should be `register` instead (always a template method in the superclass).

This I don't understand. What do you mean by "t should be `register` instead"?

Reply:
The `Executor` class define either `init` and `register` method. The latter is meant to be overridden.